core/src/main/java/org/infinispan/protostream/impl/UnknownFieldSetImpl.java

```diff
@@ -198,7 +199,7 @@ public boolean hasTag(int tag) {
 
    @Override
    public void writeExternal(ObjectOutput out) throws IOException {
-      ByteArrayOutputStreamEx baos = new ByteArrayOutputStreamEx();
+      RawByteArrayOutputStream baos = new RawByteArrayOutputStreamImpl();
       TagWriter output = TagWriterImpl.newInstance(null, baos);
       writeTo(output);
       output.flush();
```

Comment:
Lines 206-210 can be replaced with
```java
      out.writeInt(baos.getPosition());
      out.write(baos.getRawBuffer(), 0, baos.getPosition());
```

Reply:
Now that we're not exposing `getRawBuffer` this is no longer possible. Does it make more sense to wrap the provided `ObjectOutput` in an Adapter and pass this to `TagWriterImpl#newInstance`?

Reply:
`getRawBuffer` is still a public method. 
At the moment, I don't see how the adapter would work as you don't know the size before hand.